### PR TITLE
E2e subdomainless docker

### PR DESCRIPTION
**Pull request by rochoa**
This uses subdomainless configuration in the Maps API to avoid requiring a new hostname configured in `/etc/hosts`.

It also `checkout`s and `pull`s latest changes from the master branch.

### CODE REVIEW

File: DEVELOPERS.md [hunk 1]
```diff
@@ -102,8 +102,6 @@ This end to end tests cover the entire library by performing tests against real
 
 To install Docker, follow the instructions on https://docs.docker.com/install/
 
-You'll also need to add `127.0.0.1 localhost.localhost.lan` your `/etc/hosts/` file.
```
Comment (davidmanzanares) [inline DEVELOPERS.md]: Vamos!

File: test/acceptance/docker/deploy.sh [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 
 git clone https://github.com/CartoDB/Windshaft-cartodb.git ;
+git checkout master
```
Comment (davidmanzanares) [inline test/acceptance/docker/deploy.sh]: Are these really needed? A clean `clone` just was performed

Comment (rochoa) [reply test/acceptance/docker/deploy.sh]: I'm not sure. 🤷‍♂️ 

What happens when you already have a local copy?

File: DEVELOPERS.md (hunk 1 above)
Comment (VictorVelarde) [reply DEVELOPERS.md]: 👍 nice!

File: test/acceptance/docker/deploy.sh (hunk 2 above)
Comment (VictorVelarde) [reply test/acceptance/docker/deploy.sh]: (from docs) It seems that clones checks out currently active branch, by default
https://git-scm.com/docs/git-clone

Comment (davidmanzanares) [reply test/acceptance/docker/deploy.sh]: I see the intention now, but, the clone will create a new directory and the checkout and pull will be performed in the parent directory, right?

Also, do we want to enforce the checkout+pull? Isn't it useful to be able to work with a local Windshaft?

Comment (rochoa) [reply test/acceptance/docker/deploy.sh]: Actually, I just tested and it doesn't work as I was expecting. :\

If I check out an older commit than master:HEAD, expecting that the deploy script would check out master:HEAD again, but it doesn't.

Comment (rochoa) [reply test/acceptance/docker/deploy.sh]: I think it's not a problem for CI. I'm gonna remove this and push a new docker image.

I was trying to fix the situation with outdated Windshaft-cartodb repositories in local/dev environments.

**Comment by rochoa** [comment]
About the `git clone`. Do you know if that gets executed every time the image starts or it's executed and materialized when you build the image?

**Comment by davidmanzanares** [comment]
When the image starts, since we want to test Windshaft `master` every time and not a stale version. Since the `deploy.sh` is the entry point it will be called by default when running the image.

The ´clone´ would have been on the Dockerfile if it was on the build phase.

**Comment by rochoa** [comment]
Oh, stupid me! I know why it wasn't having any effect. I wasn't changing the directory where I was executing the `git checkout && git pull`. If only you would get errors. 💔

**Comment by davidmanzanares** [comment]
> the clone will create a new directory and the checkout and pull will be performed in the parent directory, right?

:joy::joy:

**Comment by davidmanzanares** [comment]
You can also open a bash session in Docker to run the commands interactively and see all the output

**Comment by VictorVelarde** [comment]
💪 vamos!

**Pull request merged**